lib/src/Uncertainty/Algorithm/OrthogonalBasis/StandardDistributionPolynomialFactory.cxx

```diff
@@ -94,60 +94,89 @@ void StandardDistributionPolynomialFactory::checkSpecificFamily()
 {
   // Check for special cases. Need a more elegant conception and implementation.
   hasSpecificFamily_ = false;
+  OrthogonalUniVariatePolynomialFamily referenceFamily;
   const String measureType(measure_.getImplementation()->getClassName());
   // Legendre factory
   if (measureType == "Uniform")
-  {
-    specificFamily_ = LegendreFactory();
-    hasSpecificFamily_ = true;
-    return;
-  }
+    {
+      referenceFamily = LegendreFactory();
+      if (referenceFamily.getMeasure() == measure_)
+	{
+	  specificFamily_ = referenceFamily;
+	  hasSpecificFamily_ = true;
+	  return;
+	}
+    }
```

Comment:
The point is to test first the class name (cheap) then the parameters (maybe costly), and since the number of parameters needed to build a specific factory depends on the distribution there is no straightforward way to simplify the code. I will propose something slightly shorter but no revolution here.
For the indentation, it is just a confusion between astyle indent (the whole buffer) and emacs indent (the current line).